annotation/lib/http.dart

```diff
@@ -9,94 +9,150 @@ class HttpMethod {
   static const String DELETE = "DELETE";
 }
 
+/// Define an API.
 @immutable
 class RestApi {
+  /// Set the API base URL.
+  ///
+  /// Base URLs should always end in `/`.
+  ///
+  /// __Correct:__
+  ///
+  /// * Base URL: `http://example.com/api/`
+  /// * Endpoint: `foo/bar/`
+  /// * Result: `http://example.com/api/foo/bar/`
+  ///
+  /// __Incorrect:__
+  ///
+  /// * Base URL: `http://example.com/api`
+  /// * Endpoint: `foo/bar/`
+  /// * Result: `http://example.com/foo/bar/`
+  ///
```

Comment:
Maybe add this?
```suggestion
  ///
  /// When you don't specify the `baseUrl`. The dio instance passed to the constructor should have it defined.
  /// Otherwise the `path` field of any `HttpMethod`  like `@POST()` should have the full URL.
```